cloudinit/config/cc_apt_configure.py

```diff
@@ -708,15 +708,15 @@ def generate_sources_list(cfg, release, mirrors, cloud):
         )
         if expected_content:
             if expected_content != util.load_text_file(apt_sources_list):
-                LOG.warning(
```

Comment:
I think there are a couple integration tests that expect exit 2 due to deb822 warnings we know are in logs. You may need to fix those up as part of this PR.

Reply:
Any clue which? I checked all of the tests that reference `NOBLE` and didn't see anything that looked like it would exit 2 due to this.

Reply:
tests/integration_tests/modules/test_cli.py::test_invalid_userdata I think anything checking for return_code == 2 could be impacted if it isn't actually setting up a known failure, but reacting to the deb822 warning